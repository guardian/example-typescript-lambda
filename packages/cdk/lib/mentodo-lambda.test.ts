import { App } from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { MentodoLambda } from './mentodo-lambda';

describe('The Mentodo stack', () => {
	it('matches the snapshot', () => {
		const app = new App();
		const stack = new MentodoLambda(app, 'MentodoLambda', {
			stack: 'playground',
			stage: 'TEST',
			env: {
				region: 'eu-west-1',
			},
		});
		const template = Template.fromStack(stack);

		/**
		 * Snapshot testing helps to understand exactly what impact a CDK change will have on the provisioned infrastructure.
		 *
		 * @see https://github.com/guardian/cdk/blob/main/docs/best-practices.md#snapshot-testing
		 */
		expect(template.toJSON()).toMatchSnapshot();
	});
});
