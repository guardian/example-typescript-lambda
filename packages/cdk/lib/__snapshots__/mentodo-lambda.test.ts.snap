// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The ExampleTypescriptLambda stack matches the snapshot 1`] = `
{
  "Metadata": {
    "gu:cdk:constructs": [
      "GuDistributionBucketParameter",
      "GuLambdaFunction",
    ],
    "gu:cdk:version": "TEST",
  },
  "Parameters": {
    "DistributionBucketName": {
      "Default": "/account/services/artifact.bucket",
      "Description": "SSM parameter containing the S3 bucket name holding distribution artifacts",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "MentodoLambda5DE055AC": {
      "DependsOn": [
        "MentodoLambdaServiceRoleDefaultPolicy907901F0",
        "MentodoLambdaServiceRole55A95968",
      ],
      "Properties": {
        "Architectures": [
          "arm64",
        ],
        "Code": {
          "S3Bucket": {
            "Ref": "DistributionBucketName",
          },
          "S3Key": "playground/TEST/mentodo-lambda/mentodo-lambda.zip",
        },
        "Environment": {
          "Variables": {
            "APP": "mentodo-lambda",
            "STACK": "playground",
            "STAGE": "TEST",
          },
        },
        "Handler": "index.main",
        "LoggingConfig": {
          "LogFormat": "JSON",
        },
        "MemorySize": 512,
        "Role": {
          "Fn::GetAtt": [
            "MentodoLambdaServiceRole55A95968",
            "Arn",
          ],
        },
        "Runtime": "nodejs20.x",
        "Tags": [
          {
            "Key": "App",
            "Value": "mentodo-lambda",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/mentodo-lambda",
          },
          {
            "Key": "Stack",
            "Value": "playground",
          },
          {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
        "Timeout": 30,
      },
      "Type": "AWS::Lambda::Function",
    },
    "MentodoLambdaServiceRole55A95968": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
        "Tags": [
          {
            "Key": "App",
            "Value": "mentodo-lambda",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/mentodo-lambda",
          },
          {
            "Key": "Stack",
            "Value": "playground",
          },
          {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "MentodoLambdaServiceRoleDefaultPolicy907901F0": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      {
                        "Ref": "DistributionBucketName",
                      },
                    ],
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      {
                        "Ref": "DistributionBucketName",
                      },
                      "/playground/TEST/mentodo-lambda/mentodo-lambda.zip",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": "ssm:GetParametersByPath",
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:aws:ssm:eu-west-1:",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":parameter/TEST/playground/mentodo-lambda",
                  ],
                ],
              },
            },
            {
              "Action": [
                "ssm:GetParameters",
                "ssm:GetParameter",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:aws:ssm:eu-west-1:",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":parameter/TEST/playground/mentodo-lambda/*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "MentodoLambdaServiceRoleDefaultPolicy907901F0",
        "Roles": [
          {
            "Ref": "MentodoLambdaServiceRole55A95968",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
  },
}
`;
